node/alphanet/src/lib.rs

```diff
@@ -3,41 +3,110 @@ extern crate serde;
 extern crate serde_derive;
 
 use std::net::{IpAddr, Ipv4Addr, SocketAddr};
-use std::sync::Arc;
+use std::sync::{Arc, RwLock};
 
 use futures::sync::mpsc::channel;
+use futures::Future;
+use tokio::runtime::Runtime;
 
 use client::Client;
 use configs::{get_alphanet_configs, ClientConfig, NetworkConfig, RPCConfig};
 use coroutines::client_task::ClientTask;
+use futures::future::IntoFuture;
+use futures::sink::Sink;
+use futures::stream::TakeWhile;
 use network::spawn_network;
 use nightshade::nightshade_task::spawn_nightshade_task;
 use primitives::types::AccountId;
+use std::thread::JoinHandle;
+use std::time::{Duration, Instant};
+use std::{panic, thread};
+use tokio::prelude::future;
+use tokio::prelude::stream::Stream;
+use tokio::sync::oneshot;
+use tokio::timer::Delay;
+use tokio_signal::unix::Signal;
 use network::proxy::ProxyHandler;
 
 pub fn start() {
     let (client_cfg, network_cfg, rpc_cfg) = get_alphanet_configs();
-    start_from_configs(client_cfg, network_cfg, rpc_cfg);
+    let mut handle = start_from_configs(client_cfg, network_cfg, rpc_cfg);
+    tokio::run(Signal::new(tokio_signal::unix::SIGINT).flatten_stream().into_future().map(drop).map_err(drop));
+    handle.shutdown();
 }
 
 pub fn start_from_configs(
     client_cfg: ClientConfig,
     network_cfg: NetworkConfig,
     rpc_cfg: RPCConfig,
-) {
+) -> ShutdownableThread {
     let client = Arc::new(Client::new(&client_cfg));
     // Use empty pipeline to launch nodes on production.
     let proxy_handlers: Vec<Arc<ProxyHandler>> = vec![];
     start_from_client(client, Some(client_cfg.account_id), network_cfg, rpc_cfg, proxy_handlers)
 }
 
+pub struct ShutdownableThread {
```

Comment:
I suggest we extract `ShutfownableThread` into a separate file, e.g. `shutdownable_tokio.rs`.